internal/plugins/src/main/kotlin/org/jsonschema2dataclass/internal/plugin/publishing/publishing.kt

```diff
@@ -0,0 +1,116 @@
+package org.jsonschema2dataclass.internal.plugin.publishing
+
+import basePluginExtension
+import groovy.util.Node
+import groovy.util.NodeList
+import org.gradle.api.Project
+import org.gradle.api.publish.PublishingExtension
+import org.gradle.api.publish.maven.MavenPublication
+import org.gradle.kotlin.dsl.configure
+import org.gradle.kotlin.dsl.get
+import org.gradle.kotlin.dsl.getByName
+import org.gradle.kotlin.dsl.register
+import org.gradle.plugins.signing.SigningExtension
+
+fun applyPublishing(project: Project, signing: Boolean = false) {
+    project.plugins.apply("maven-publish")
+
+    project.configure<PublishingExtension> {
+        publications {
+            register<MavenPublication>("release") release@{
+                if (signing) {
+                    project.configure<SigningExtension> {
+                        sign(this@release)
+                    }
+                }
+                setupModuleIdentity(project)
+                from(project.components["java"])
+                setupLinks()
+            }
+        }
+    }
+    project.afterEvaluate {
+        project.configure<PublishingExtension> {
+            publications {
+                getByName<MavenPublication>("release") {
+                    reorderNodes(project)
+                }
+            }
+        }
+    }
+}
+
+private fun MavenPublication.setupModuleIdentity(project: Project) {
+    project.afterEvaluate {
+        artifactId = project.basePluginExtension.archivesName.get()
```

Comment:
> Why do you rename artifact id in your plugin?

I change the archivesName in each build.gradle because I don't like the default which is the module name. "plugin.jar" and "compat.jar" are meaningless names and I don't want to see them on the classpath. Adding the prefix to the name makes the jar names consumable even when the group name is not present, see the highlights here:
![image](https://user-images.githubusercontent.com/2906988/224508230-aa5bb640-33ab-4abb-b464-bf10cbfe1fc9.png)

I used archivesName to change this for the build/libs/*.jar, and I pull the value from there into the artifactId to make the invariant that the jar name is the same as the Maven artifactId.

Reply:
I understand and agree why to rename a supporting library. I need to look on this from perspective of an application developer. Thank you